engine_test.go

```diff
@@ -2209,6 +2209,63 @@ func TestReplaceIntoErrors(t *testing.T) {
 	}
 }
 
+func TestUpdate(t *testing.T) {
+	var updates = []struct {
+		updateQuery    string
+		expectedUpdate []sql.Row
+		selectQuery     string
+		expectedSelect  []sql.Row
+	}{
+		{
+			"UPDATE mytable SET s = 'updated row' WHERE i = 1;",
+			[]sql.Row{{int64(1), int64(1)}},
+			"SELECT * FROM mytable WHERE i = 1;",
+			[]sql.Row{{int64(1), "updated row"}},
+		},
+		{
+			"UPDATE mytable SET s = 'updated row' WHERE i <> 9999;",
+			[]sql.Row{{int64(3), int64(3)}},
+			"SELECT * FROM mytable;",
+			[]sql.Row{{int64(1), "updated row"},{int64(2), "updated row"},{int64(3), "updated row"}},
+		},
+		{
+			"UPDATE floattable SET f32 = f32 + f32, f64 = f32 * f64 WHERE i = 2;",
+			[]sql.Row{{int64(1), int64(1)}},
+			"SELECT * FROM floattable WHERE i = 2;",
+			[]sql.Row{{int64(2), float32(3.0), float64(4.5)}},
```

Comment:
This isn't the right result and indicates you have a bug. The second value should be 1.5 * 1.5. Looks like your column updates aren't independent.

I would write some additional tests for setting multiple columns in the same updates statement once you fix this.

Reply:
I stand corrected:

https://dev.mysql.com/doc/refman/8.0/en/update.html

I guess we do what MySQL does, even though it's wrong :(

Reply:
_Non-standard_ 😄